Next

mich181189 opened the pull request:
Some fixes for you. Mostly fixes. The cmakey one improves the errors generated when you are missing some packages

mich181189 committed:
libdumpalloc: Fix deadlock when memalign is called

memalign (at least in uclibc) calls malloc after taking a mutex lock.
This was causing a deadlock between the mutex in dumpalloc and the mutex
in uclibc. By intercepting memalign calls and taking our lock, we ensure
our lock is acquired before libc's lock, and therefore prevent this deadlock.

mich181189 committed:
Fixed a small typo in Readme.md

The netcat command was missing the p option in one place.
This commit fixes the documentation

jpsadler commented:
Hi Michael,

Small thing:  could you split the readalloc and libdumpalloc changes, and prefix the commit synopsis with "readalloc" or "libdumpalloc" accordingly?

I've (mostly) done that on my "next" branch, but not 100% consistent unfortunately...

Thanks,

John.

jpsadler commented on libdumpalloc/CMakeLists.txt:
```diff
@@ -36,7 +36,7 @@ target_link_libraries ( dumpalloc-python "${PYTHON_LIBRARIES}" )
 endif()
 
-add_subdirectory ( manual-test-programs )
+#add_subdirectory ( manual-test-programs )
```
Did you mean to comment this out?

mich181189 replied:
Yes. manual-test-programs doesn't exist there, so this throws an error.

maybe you have some local files that aren't checked in?

Commit:
libdumpalloc: comment out missing directory from CMakeLists.txt

the directory manual-test-programs does not seem to exist, and so
causes an error while building.

This commit comments it out so CMake doesn't complain.

Commit:
readalloc: Fix up CMake file

A few improvements to the CMakeLists.txt file - notably specifying
Boost components, so you get cmake-time errors if your system is
missing required dependencies, and making included more CMake-like

mich181189 commented:
This should be a bit tidier now. I've labeled the commits and reworked a couple of others to split them by area.
